src/Npgsql/NpgsqlBinaryImporter.cs

```diff
@@ -154,6 +154,8 @@ public Task WriteAsync<T>([AllowNull] T value, CancellationToken cancellationTok
 
         Task Write<T>([AllowNull] T value, bool async)
         {
+            CheckColumnInTable(_column, NumColumns);
+
             var p = _params[_column];
```

Comment:
@roji As you can see `Write` happens after access to the parameter array and moving check into it won't help.

Reply:
Yeah @roji, like @YohDeadfall mentioned moving the check to the private `Write` is too late. The overloads themselves already access the `_params` array.